src/nyc_trees/js/src/lib/storage.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+var $ = require('jquery'),
+    storage = window.localStorage || {};
+
+
+function Storage(options) {
+    var opts = $.extend({}, {
+        key: '',
+        getState: $.noop,
+        validate: $.noop
+    }, options);
+
+    function save() {
+        storage[opts.key] = serialize();
+    }
+
+    function load() {
+        var state = deserialize(storage[opts.key]);
+        if (state) {
+            opts.validate(state);
```

Comment:
Letting the exceptions from `validate` bubble up isn't the best thing...

Technically the `LocalStorage` could be edited by a user (though it seems like a low probability), or a bug in our code could make saved state no longer validate.

It's probably more robust to wrap this in a try/catch, and if there are any exceptions, log it and delete whatever was saved.